WordPress/Classes/Networking/ReaderPostServiceRemote.m

```diff
@@ -266,45 +287,121 @@ - (RemoteReaderPost *)formatPostDictionary:(NSDictionary *)dict
+    post.postID = [dict numberForKey:PostRESTKeyID];
+    post.postTitle = [self stringOrEmptyString:[dict stringForKey:PostRESTKeyTitle]];
+    post.siteID = [dict numberForKey:PostRESTKeySiteID];
     post.sortDate = [self sortDateFromPostDictionary:dict];
-    post.status = [self stringOrEmptyString:[dict stringForKey:@"status"]];
-    post.summary = [self stringOrEmptyString:[dict stringForKey:@"excerpt"]];
+    post.status = [self stringOrEmptyString:[dict stringForKey:PostRESTKeyStatus]];
+    post.summary = [self stringOrEmptyString:[dict stringForKey:PostRESTKeyExcerpt]];
     post.tags = [self tagsFromPostDictionary:dict];
-    post.isSharingEnabled = [[dict numberForKey:@"sharing_enabled"] boolValue];
-    post.isLikesEnabled = [[dict numberForKey:@"likes_enabled"] boolValue];
+    post.isSharingEnabled = [[dict numberForKey:PostRESTKeySharingEnabled] boolValue];
+    post.isLikesEnabled = [[dict numberForKey:PostRESTKeyLikesEnabled] boolValue];
+
+    NSDictionary *tags = [self primaryAndSecondaryTagsFromPostDictionary:dict];
+    if (tags) {
+        post.primaryTag = [tags stringForKey:TagKeyPrimary];
+        post.primaryTagSlug = [tags stringForKey:TagKeyPrimarySlug];
+        post.secondaryTag = [tags stringForKey:TagKeySecondary];
+        post.secondaryTagSlug = [tags stringForKey:TagKeySecondarySlug];
+    }
+
+    post.isExternal = [[dict numberForKey:PostRESTKeyIsExternal] boolValue];
+    post.isJetpack = [[dict numberForKey:PostRESTKeyIsJetpack] boolValue];
+    post.wordCount = [dict numberForKey:PostRESTKeyWordCount];
+    post.readingTime = [self readingTimeForWordCount:post.wordCount];
+
     if ([dict arrayForKeyPath:@"discover_metadata.discover_fp_post_formats"]) {
-        post.sourceAttribution = [self sourceAttributionFromDictionary:[dict dictionaryForKey:@"discover_metadata"]];
+        post.sourceAttribution = [self sourceAttributionFromDictionary:[dict dictionaryForKey:PostRESTKeyDiscoverMetadata]];
     }
     return post;
 }
 
+- (NSDictionary *)primaryAndSecondaryTagsFromPostDictionary:(NSDictionary *)dict
+{
+    NSString *primaryTag = @"";
+    NSString *primaryTagSlug = @"";
+    NSString *secondaryTag = @"";
+    NSString *secondaryTagSlug = @"";
+    NSString *editorialTag;
+    NSString *editorialSlug;
+
+    NSArray *remoteTags = [[dict dictionaryForKey:PostRESTKeyTags] allValues];
+    if (remoteTags) {
+        NSInteger postCount = 0;
+
+        for (NSDictionary *tag in remoteTags) {
+            NSInteger count = [[tag numberForKey:PostRESTKeyPostCount] integerValue];
+            if (count > postCount) {
+                secondaryTag = primaryTag;
+                secondaryTagSlug = primaryTagSlug;
+
+                primaryTag = [tag stringForKey:PostRESTKeyName] ?: @"";
+                primaryTagSlug = [tag stringForKey:PostRESTKeySlug] ?: @"";
+
+                postCount = count;
+            }
+        }
+    }
```

Comment:
Sort of.  The logic should be:
- Loop over all the tags.
- If the current tag's post count is greater than the previous post count, make it the new primary tag, and make a previous primary tag the secondary tag. 

We're only concerned with the two tags with the highest post count.
I'll add a comment for clarity.